src/ZigUP/Source/global.c

```diff
@@ -15,4 +16,31 @@ volatile uint32 S0=0;
 volatile uint8 STATE_LIGHT=0;
 volatile uint8 STATE_LED=0;
 
-uint8 TEMP_SENSOR=0;
+int8 TEMP_SENSOR=0;
+
+uint8 ds18b20_ROM[8]; // ROM Bit
+uint8 ds18b20_lastDiscrep = 0; // last discrepancy
+uint8 ds18b20_doneFlag = 0; // Done flag
+uint8 ds18b20_FoundROM[ds18b20_MAX_DEVICES][8]; // table of found ROM codes
+uint8 ds18b20_numROMs;
+uint8 ds18b20_dowcrc;
+uint8 ds18b20_current;
+volatile uint8 ds18b20_dscrc_table[] = {
+0, 94,188,226, 97, 63,221,131,194,156,126, 32,163,253, 31, 65,
+157,195, 33,127,252,162, 64, 30, 95, 1,227,189, 62, 96,130,220,
+35,125,159,193, 66, 28,254,160,225,191, 93, 3,128,222, 60, 98,
+190,224, 2, 92,223,129, 99, 61,124, 34,192,158, 29, 67,161,255,
+70, 24,250,164, 39,121,155,197,132,218, 56,102,229,187, 89, 7,
+219,133,103, 57,186,228, 6, 88, 25, 71,165,251,120, 38,196,154,
+101, 59,217,135, 4, 90,184,230,167,249, 27, 69,198,152,122, 36,
+248,166, 68, 26,153,199, 37,123, 58,100,134,216, 91, 5,231,185,
+140,210, 48,110,237,179, 81, 15, 78, 16,242,172, 47,113,147,205,
+17, 79,173,243,112, 46,204,146,211,141,111, 49,178,236, 14, 80,
+175,241, 19, 77,206,144,114, 44,109, 51,209,143, 12, 82,176,238,
+50,108,142,208, 83, 13,239,177,240,174, 76, 18,145,207, 45,115,
+202,148,118, 40,171,245, 23, 73, 8, 86,180,234,105, 55,213,139,
+87, 9,235,181, 54,104,138,212,149,203, 41,119,244,170, 72, 22,
+233,183, 85, 11,136,214, 52,106, 43,117,151,201, 74, 20,246,168,
+116, 42,200,150, 21, 75,169,247,182,232, 10, 84,215,137,107, 53};
+char EXT_Temperature_string[1 + 8*2 + 1 + 7 + 1] = "\x05Empty";

```

Comment:
This doesn't "overflow":
`char EXT_Temperature_string[1 + 8*2 + 1 + 7 + 1] = { 5, 'E', 'm', 'p', 't', 'y', 0 };`

Reply:
With that, both CC2530 and CC2531 works and don't crash.
Do you have an idea what exactly was the problem there?
PS: Maybe we should change that "Empty" to something else, when you split that string by ":" later in zigbee2mqtt converters.

Reply:
Great finding: 
Seems that i did nor known what kinda code compiler does:
"\x05Empty"; -> 5^mpty 
{ 5, 'E', 'm', 'p', 't', 'y', 0 }; ->  Empty

The default value should be "0000000000000000:-999.99"
I will fix it soon.

Reply:
I made previous changes and testing it with 16 sensors. Sensors seem to work fine but now i have other problem that ZigUP does not want to pair with coordinator. ZigUP is on reset loop (as it has designed if it cannot connect to coordinator) and zigbee2mqtt logs say: "PM Cannot get the Node Descriptor of the Device: 0x00124b001bd816a1 (Error: Timed out after 10000 ms)" after ZigUP message. 
I also downgraded firmware to vanilla ZigUP without any my modification what has worked before but also this FW cannot make a connection.
I have reseted zigbee2mqtt, mosquito, ZigUP stick and powering off same time coordinator stick. All traces from ZigUP have cleared from zigbee2mqtt configuration and database file.

I try to check tomorrow night what causing this.

Reply:
I left ZigUP powers on last night and seems tht after 4hours ans 3mins it manage pairing it self. I have a rather big zigbee network and between coordinator and ZigUP there is 2 routers (about 50 devices total)

Reply:
I know that behaviour. But even with other commercial devices.

Reply:
Then it does not need to check it forward currently.

Sniffer shows proper data so my behalf it's ready to merge into master.
```
Frame 170: 126 bytes on wire (1008 bits), 124 bytes captured (992 bits) on interface 0
IEEE 802.15.4 Data, Dst: 0x5452, Src: 0x0570
    Frame Control Field: 0x8861, Frame Type: Data, Acknowledge Request, PAN ID Compression, Destination Addressing Mode: Short/16-bit, Frame Version: IEEE Std 802.15.4-2003, Source Addressing Mode: Short/16-bit
    Sequence Number: 156
    Destination PAN: 0x1a62
    Destination: 0x5452
    Source: 0x0570
    [Extended Source: TexasIns_00:1b:d8:16:a1 (00:12:4b:00:1b:d8:16:a1)]
    [Origin: 54]
ZigBee Network Layer Data, Dst: 0x0000, Src: 0x0570
    Frame Control Field: 0x0248, Frame Type: Data, Discover Route: Enable, Security Data
    Destination: 0x0000
    Source: 0x0570
    Radius: 30
    Sequence Number: 73
    [Extended Source: TexasIns_00:1b:d8:16:a1 (00:12:4b:00:1b:d8:16:a1)]
    [Origin: 54]
    ZigBee Security Header
ZigBee Application Support Layer Data, Dst Endpt: 1, Src Endpt: 8
    Frame Control Field: Data (0x00)
    Destination Endpoint: 1
    Cluster: On/Off (0x0006)
    Profile: Home Automation (0x0104)
    Source Endpoint: 8
    Counter: 0
ZigBee Cluster Library Frame, Command: Report Attributes, Seq: 0
    Frame Control Field: Profile-wide (0x00)
    Sequence Number: 0
    Command: Report Attributes (0x0a)
    Attribute Field
    Attribute Field, Float: 1.#QNAN
    Attribute Field, Float: 1.#QNAN
    Attribute Field, Float: 1.#QNAN
    Attribute Field, Uint32: 0
    Attribute Field, Float: 1.#QNAN
    Attribute Field, Uint16: 0
    Attribute Field, Uint16: 0
    Attribute Field, String: 0000000000000000:-999.99
        Attribute: Unknown (0xa198)
        Data Type: Character String (0x42)
        String: 0000000000000000:-999.99

```